ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -130,7 +133,8 @@ public void processResponse(RestRequest restRequest, RestResponseChannel respons
       if (operationOrBlobId.startsWith("/")) {
         operationOrBlobId = operationOrBlobId.substring(1);
       }
-      if (blobInfo == null && !restRequest.getRestMethod().equals(RestMethod.OPTIONS)) {
+      if (blobInfo == null && !restRequest.getRestMethod().equals(RestMethod.OPTIONS) && !restRequest.getRestMethod()
```

Comment:
Minor: I'd move up the line that extracts restMethod into a variable and use that here.